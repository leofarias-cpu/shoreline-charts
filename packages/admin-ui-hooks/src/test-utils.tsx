import 'jest-location-mock'

export function jestWindowMock() {
  let history: string[] = [window.location.pathname]
  let index = 0
  let popstateCallback = () => {}

  Object.defineProperties(window.history, {
    pushState: {
      value: (data: { path: string }) => {
        history = [...history.splice(0, index + 1), data.path]
        index += 1

        window.location.assign(data.path)
        popstateCallback()
      },
    },
    back: {
      value: () => {
        if (index <= 0) return

        index -= 1
        window.location.assign(history[index])
        popstateCallback()
      },
    },
    forward: {
      value: () => {
        if (index >= history.length) return

        index += 1
        window.location.assign(history[index])
        popstateCallback()
      },
    },
  })
  Object.defineProperties(window, {
    addEventListener: {
      value: (name: string, fn: () => void) => {
        if (name === 'popstate') {
          popstateCallback = fn
        }
      },
    },
  })
}

export function setQueryParam(query: Record<string, string | undefined>) {
  const queryParams = new URLSearchParams(window.location.search)

  Object.entries(query).forEach((element: [string, string | undefined]) => {
    if (!element[1]) {
      queryParams.delete(element[0])

      return
    }

    queryParams.set(element[0], element[1])
  })

  window.location.assign(`?${queryParams.toString()}`)
}

export function getCurrentQuery() {
  const queryParams = new URLSearchParams(window.location.search)

  return Object.fromEntries(queryParams.entries())
}

export function cleanQueryParam() {
  window.location.assign('')
}
