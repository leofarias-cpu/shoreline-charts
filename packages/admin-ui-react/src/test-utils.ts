import type { ReactElement } from 'react'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { ThemeProvider } from './system'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export { render }
