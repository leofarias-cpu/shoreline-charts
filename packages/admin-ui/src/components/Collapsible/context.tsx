import React, { useContext, createContext } from 'react'
import type { DisclosureProps } from 'reakit'
import invariant from 'tiny-invariant'

interface CollapsibleContextProps extends DisclosureProps {
  variant: { container: string; header: string; content: string }
}

const CollapsibleContext = createContext<CollapsibleContextProps | null>(null)

export function useCollapsibleContext() {
  const context = useContext(CollapsibleContext)

  invariant(
    context,
    `Do not use Collapsible's composites outside of Collapsible context`
  )

  return context
}

export function CollapsibleProvider({
  children,
  ...restProps
}: CollapsibleContextProps) {
  return (
    <CollapsibleContext.Provider value={{ ...restProps }}>
      {children}
    </CollapsibleContext.Provider>
  )
}

interface TreeContextProps {
  isRoot: boolean
}

const TreeContext = createContext<TreeContextProps>({ isRoot: true })

export function useTree() {
  const { isRoot } = useContext(TreeContext)

  return { isRoot }
}

export function TreeProvider(props: React.PropsWithChildren<TreeContextProps>) {
  const { children, ...restProps } = props

  return (
    <TreeContext.Provider value={{ ...restProps }}>
      {children}
    </TreeContext.Provider>
  )
}
