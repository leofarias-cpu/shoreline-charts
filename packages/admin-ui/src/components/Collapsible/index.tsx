import type { PropsWithoutRef, ReactNode } from 'react'
import React from 'react'
import type { DisclosureProps } from 'reakit/Disclosure'
import {
  useDisclosureState,
  Disclosure as ReakitDisclosure,
  DisclosureContent,
  DisclosureStateReturn,
} from 'reakit/Disclosure'
import { IconCaret } from '@vtex/admin-ui-icons'
import { motion, AnimateSharedLayout, AnimatePresence } from 'framer-motion'
import { Flex } from '@vtex/admin-primitives'

import {
  CollapsibleProvider,
  useCollapsibleContext,
  useRootContext,
  RootContext,
} from './context'
import { Button } from '../Button'
import type { SystemComponent, SystemComponentProps } from '../../types'
import { useGroup } from '../Group'
import { tag } from '@vtex/admin-ui-react'

/**
 * A Collapsible is a container that allows toggling the display of content. It can be nested as well.
 * @example
 * ```jsx
 * import { Collapsible, useCollapsible } from `@vtex/admin-ui`
 * const state = useCollapsible()
 * <Collapsible state={state}>
 *   <Collapsible.Header label="Title goes here">
 *     {children}
 *   </Collapsible.Header>
 *   <Collapsible.Content>{content}</Collapsible.Content>
 * </Collapsible>
 * ```
 */
export function Collapsible(props: CollapsibleProps) {
  const isRoot = useRootContext()
  const { grouped } = useGroup()

  const { csx, children, disabled, focusable, state, ...collapsibleProps } =
    props

  const reakitProps = {
    focusable,
    disabled,
    ...state,
  }

  return (
    <AnimateSharedLayout>
      <tag.div
        as={motion.div}
        layout
        csx={{
          bg: 'light.primary',
          borderColor: 'mid.tertiary',
          borderStyle: 'solid',
          borderWidth: grouped ? 0 : 1,
          borderRadius: 'default',
          '.__admin-ui-collapsible--header': {
            padding: !isRoot ? 4 : 6,
            paddingLeft: !isRoot ? 0 : 2,
          },
          '.__admin-ui-collapsible--content': {
            paddingX: !isRoot ? 4 : 6,
            paddingBottom: !isRoot ? 4 : 6,
          },
          ...csx,
        }}
        {...collapsibleProps}
      >
        <CollapsibleProvider {...reakitProps}>
          <RootContext.Provider value={false}>{children}</RootContext.Provider>
        </CollapsibleProvider>
      </tag.div>
    </AnimateSharedLayout>
  )
}

export function Header(props: CollapsibleHeaderProps) {
  const { children, label, csx, ...headerProps } = props

  return (
    <tag.header
      as={motion.header}
      layout
      className="__admin-ui-collapsible--header"
      csx={{
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'space-between',
        'div > button:nth-of-type(n+2)': {
          marginLeft: 1,
        },
        ...csx,
      }}
      {...headerProps}
    >
      <Disclosure>{label}</Disclosure>
      <Flex>{children}</Flex>
    </tag.header>
  )
}

function Disclosure({ children }: { children: ReactNode }) {
  const { visible, ...disclosureProps } = useCollapsibleContext()

  return (
    <ReakitDisclosure visible={visible} {...disclosureProps}>
      {(enhancedProps) => {
        return (
          <Button
            {...enhancedProps}
            iconPosition="start"
            icon={
              <IconCaret
                csx={{
                  transition: 'transform 150ms ease',
                  transform: `rotate(${visible ? 180 : 90}deg)`,
                }}
              />
            }
            variant="tertiary"
            csx={{
              color: 'dark.primary',
              textTransform: 'none',
              '&:hover': { backgroundColor: 'transparent' },
              '&:active': { backgroundColor: 'transparent' },
            }}
          >
            {children}
          </Button>
        )
      }}
    </ReakitDisclosure>
  )
}

export function Content(props: CollapsibleContentProps) {
  const { children, csx, ...contentProps } = props
  const disclosureProps = useCollapsibleContext()

  return (
    <DisclosureContent {...disclosureProps}>
      {({ style, ...enhancedProps }) => {
        return (
          <AnimatePresence>
            {disclosureProps.visible && (
              <motion.section
                layout
                initial="hidden"
                animate="visible"
                exit="hidden"
                variants={{
                  visible: {
                    transition: {
                      duration: 0.15,
                    },
                    height: 'auto',
                    opacity: 1,
                  },
                  hidden: {
                    transition: {
                      duration: 0.15,
                    },
                    height: 0,
                    opacity: 0,
                  },
                }}
                {...(enhancedProps as PropsWithoutRef<'section'>)}
              >
                <tag.div
                  className="__admin-ui-collapsible--content"
                  csx={{ ...csx }}
                  {...contentProps}
                >
                  {children}
                </tag.div>
              </motion.section>
            )}
          </AnimatePresence>
        )
      }}
    </DisclosureContent>
  )
}

/**
 * ```
 * Collapsible.Header is always visible.
 * Disclosure Button -> always on the left side, and is responsible for controlling the content visibility.
 * Actions Panel -> always on the right side.
 * ```
 */
Collapsible.Header = Header
/**
 * ```
 * Collapsible content.
 * Can be visible or hidden.
 * ```
 */
Collapsible.Content = Content

export interface CollapsibleProps
  extends SystemComponentProps<
    Pick<DisclosureProps, 'focusable' | 'disabled' | 'children'>
  > {
  /** useCollapsible hook return */
  state: DisclosureStateReturn
}

export interface CollapsibleHeaderProps extends SystemComponent {
  /**
   * Disclosure Button label
   */
  label?: ReactNode
  /**
   * Actions panel
   */
  children?: ReactNode
}

export interface CollapsibleContentProps extends SystemComponent {
  children?: ReactNode
}

export { useDisclosureState as useCollapsible }
export { DisclosureStateReturn }
