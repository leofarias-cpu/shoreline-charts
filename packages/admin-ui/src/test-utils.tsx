import React, { useRef } from 'react'
import { axe } from 'jest-axe'
import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { ThemeProvider } from '@vtex/admin-core'
import type { ReactElement } from 'react'
import { renderHook, act } from '@testing-library/react-hooks'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

/**
 * HOC to wrap a component with its state hook
 * @example
 * // Cmp has the useCmp state hook
 * const StatefulCpm = withState(Cmp, () => useCmpState({ isSelected: false }))
 */
function withState(Component: any, useStateHook: () => any) {
  function StatefulComponent(props: any) {
    // Avoids an invalid hook call
    const ref = useRef(useStateHook())

    return <Component state={ref.current} {...props} />
  }

  StatefulComponent.displayName = Component.displayName

  return StatefulComponent
}

function jestMatchMedia() {
  /**
   * ðŸš§ Workaround for window.match media
   * @see https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
   */
  return Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(), // Deprecated
      removeListener: jest.fn(), // Deprecated
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  })
}

export * from '@testing-library/react'
export { render, withState, axe, jestMatchMedia, renderHook, act }
