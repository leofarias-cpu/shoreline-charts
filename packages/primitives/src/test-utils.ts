import type { RenderOptions } from '@testing-library/react'
import { render as baseRender } from '@testing-library/react'
import { ThemeProvider } from '@vtex/admin-core'
import type { ReactElement } from 'react'

function render(ui: ReactElement, options?: Omit<RenderOptions, 'queries'>) {
  return baseRender(ui, { wrapper: ThemeProvider, ...options })
}

export * from '@testing-library/react'
export { render }
