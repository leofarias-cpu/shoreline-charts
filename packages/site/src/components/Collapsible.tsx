import type { ReactNode } from 'react'
import React from 'react'
import {
  Collapsible,
  CollapsibleHeader,
  CollapsibleContent,
  useCollapsibleState,
} from '@vtex/admin-ui'

interface Props {
  heading: string
  component: string
  children: ReactNode
  visible?: boolean
}

export default function DocsCollapsbile(props: Props) {
  const { heading, component, children, visible, ...restProps } = props
  const collapsible = useCollapsibleState({ visible })

  return (
    <Collapsible state={collapsible} {...restProps} csx={{ marginBottom: 2 }}>
      <CollapsibleHeader label={heading} />
      <CollapsibleContent>{children}</CollapsibleContent>
    </Collapsible>
  )
}
