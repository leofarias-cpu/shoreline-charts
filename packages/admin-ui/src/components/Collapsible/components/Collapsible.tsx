import React from 'react'
import type { DisclosureProps, DisclosureStateReturn } from 'reakit/Disclosure'

import { jsx } from '@vtex/admin-ui-react'

import { CollapsibleProvider } from '../context'
import { useGroup } from '../../Group'

/**
 * A Collapsible is a container that allows toggling the display of content. It can be nested as well.
 * @example
 * ```jsx
 * import { Collapsible, useCollapsibleState } from `@vtex/admin-ui`
 * const state = useCollapsibleState()
 * <Collapsible state={state}>
 *   <Collapsible.Header label="Title goes here">
 *     {children}
 *   </Collapsible.Header>
 *   <Collapsible.Content>{content}</Collapsible.Content>
 * </Collapsible>
 * ```
 */
export const Collapsible = jsx('div')(
  {
    bg: 'light.primary',
    borderColor: 'mid.tertiary',
    borderStyle: 'solid',
    borderWidth: 1,
    borderRadius: 'default',
  },
  {
    options: ['disabled', 'focusable', 'state'],
    useOptions(options: CollapsibleOptions, props) {
      const { disabled, focusable, state } = options
      const { csx, children, ...collapsibleProps } = props
      const { grouped } = useGroup()

      return {
        ...collapsibleProps,
        csx: {
          borderWidth: grouped ? 0 : 1,
          ...csx,
        },
        children: (
          <CollapsibleProvider
            focusable={focusable}
            disabled={disabled}
            {...state}
          >
            {children}
          </CollapsibleProvider>
        ),
      }
    },
  }
)

export interface CollapsibleOptions
  extends Pick<DisclosureProps, 'focusable' | 'disabled'> {
  /** useCollapsibleState hook return */
  state: DisclosureStateReturn
}
