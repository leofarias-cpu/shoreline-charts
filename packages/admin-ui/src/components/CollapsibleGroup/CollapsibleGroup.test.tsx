import React from 'react'
import { render } from '../../test-utils'

import { CollapsibleGroup } from './index'
import type { DisclosureStateReturn } from '../Collapsible'
import { useCollapsible, Collapsible } from '../Collapsible'

function CollapsibleState({
  children,
}: {
  children: (state: DisclosureStateReturn) => JSX.Element
}) {
  const state = useCollapsible({ baseId: 'mocked-id' })

  return children(state)
}

describe('CollapsibleGroup tests', () => {
  it('should have overridable styles', () => {
    const { getByTestId } = render(
      <CollapsibleGroup data-testid="collapsible-group" csx={{ bg: 'azure' }}>
        <CollapsibleState>
          {(state) => (
            <Collapsible state={state}>
              <Collapsible.Header label="Collapsible Header" />
              <Collapsible.Content>Collapsible Content</Collapsible.Content>
            </Collapsible>
          )}
        </CollapsibleState>
      </CollapsibleGroup>
    )

    expect(getByTestId('collapsible-group')).toHaveStyleRule(
      'background-color',
      'azure'
    )
  })

  it('should match snapshot', () => {
    const { asFragment } = render(
      <CollapsibleGroup>
        <CollapsibleState>
          {(state) => (
            <Collapsible state={state}>
              <Collapsible.Header label="Collapsible Header" />
              <Collapsible.Content>Collapsible Content</Collapsible.Content>
            </Collapsible>
          )}
        </CollapsibleState>
      </CollapsibleGroup>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should not render invalid children', () => {
    const { asFragment } = render(
      <CollapsibleGroup>
        {undefined}
        <CollapsibleState>
          {(state) => (
            <Collapsible state={state}>
              <Collapsible.Header label="Collapsible Header" />
              <Collapsible.Content>Collapsible Content</Collapsible.Content>
            </Collapsible>
          )}
        </CollapsibleState>
        {null}
      </CollapsibleGroup>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
