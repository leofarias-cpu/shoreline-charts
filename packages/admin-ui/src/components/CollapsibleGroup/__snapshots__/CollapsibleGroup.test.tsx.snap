// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleGroup tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-6 {
  background-color: #FFFFFF;
  border-radius: 4px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 1.5rem;
  padding-left: 0.5rem;
}

.emotion-5 div>button:nth-of-type(n+2) {
  margin-left: 0.25rem;
}

.emotion-3 {
  font-family: VTEXTrustVF,-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  background-color: transparent;
  color: #323845;
  font-size: 0.875rem;
  height: 40px;
  width: auto;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
  text-transform: none;
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-3:hover {
  color: #214491;
  background-color: rgba(202,215,241,0.24);
}

.emotion-3:active {
  color: #1a3470;
  background-color: rgba(186,203,236,0.32);
}

.emotion-3:disabled {
  color: #898F9E;
}

.emotion-3 svg {
  width: 24px;
  min-width: 24px;
  height: 24px;
  min-height: 24px;
  margin: 0.25rem;
}

.emotion-3:hover {
  background-color: transparent;
}

.emotion-3:active {
  background-color: transparent;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-0 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: -webkit-transform 150ms ease;
  transition: transform 150ms ease;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-7 {
  background-color: #FFFFFF;
  color: #323845;
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  border-color: #E0E2E7;
  padding: 0rem;
}

<div
    class="emotion-7"
  >
    <div
      class="emotion-6"
    >
      <header
        class="emotion-5"
      >
        <button
          aria-controls="mocked-id"
          aria-expanded="false"
          class="emotion-3"
          type="button"
        >
          <div
            class="emotion-2"
          >
            <div
              class="emotion-1"
            >
              <svg
                class="emotion-0"
                fill="none"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 14L12 10L8 14"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
               Collapsible Header
            </div>
          </div>
        </button>
        <div
          class="emotion-4"
        />
      </header>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CollapsibleGroup tests should not render invalid children 1`] = `
<DocumentFragment>
  .emotion-6 {
  background-color: #FFFFFF;
  border-radius: 4px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  padding: 1.5rem;
  padding-left: 0.5rem;
}

.emotion-5 div>button:nth-of-type(n+2) {
  margin-left: 0.25rem;
}

.emotion-3 {
  font-family: VTEXTrustVF,-apple-system,system-ui,BlinkMacSystemFont,sans-serif;
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  cursor: pointer;
  position: relative;
  background-color: transparent;
  color: #323845;
  font-size: 0.875rem;
  height: 40px;
  width: auto;
  padding-left: 0.5rem;
  padding-right: 0.75rem;
  text-transform: none;
}

.emotion-3:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-3:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-3:hover {
  color: #214491;
  background-color: rgba(202,215,241,0.24);
}

.emotion-3:active {
  color: #1a3470;
  background-color: rgba(186,203,236,0.32);
}

.emotion-3:disabled {
  color: #898F9E;
}

.emotion-3 svg {
  width: 24px;
  min-width: 24px;
  height: 24px;
  min-height: 24px;
  margin: 0.25rem;
}

.emotion-3:hover {
  background-color: transparent;
}

.emotion-3:active {
  background-color: transparent;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  width: 100%;
  margin: auto;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  visibility: visible;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-0 {
  min-width: 24px;
  min-height: 24px;
  width: 24px;
  height: 24px;
  -webkit-transform: rotate(90deg);
  -moz-transform: rotate(90deg);
  -ms-transform: rotate(90deg);
  transform: rotate(90deg);
  -webkit-transition: -webkit-transform 150ms ease;
  transition: transform 150ms ease;
}

.emotion-4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-7 {
  background-color: #FFFFFF;
  color: #323845;
  border-width: 1px;
  border-style: solid;
  border-radius: 4px;
  border-color: #E0E2E7;
  padding: 0rem;
}

<div
    class="emotion-7"
  >
    <div
      class="emotion-6"
    >
      <header
        class="emotion-5"
      >
        <button
          aria-controls="mocked-id"
          aria-expanded="false"
          class="emotion-3"
          type="button"
        >
          <div
            class="emotion-2"
          >
            <div
              class="emotion-1"
            >
              <svg
                class="emotion-0"
                fill="none"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16 14L12 10L8 14"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
               Collapsible Header
            </div>
          </div>
        </button>
        <div
          class="emotion-4"
        />
      </header>
    </div>
  </div>
</DocumentFragment>
`;
